### DISCUSSION TITLE
Debugging connection attempt timeouts

### DISCUSSION THREAD

**Discussion by fschroder-slyp** [psycopg/psycopg2#1370] [Q&A] [answered]
Hi, 
I'm trying to debug pg connection attempt timeouts in our environment.
Over 99.5% attempts succeed very quickly, but the others fail after 5s.
I use AWS, the code is running in a lambda trying to connect to RDS. 
To be able to debug it, I need to log as much information as possible about the failed attempt. 
This means trying to get the local port being used to create the connection and the actual pg instance it is trying to connect to (the RDS endpoint has many readers behind it).

After a failed connection attempt I get an OperationalError which doesn't provide much information. I suspect psycopg can't really provide this kind of information, but perhaps someone can provide some python lower level suggestions to debug the issue.

Thanks,
Federico

**Comment by dvarrazzo** [comment, marked as answer]
If you want to see something more of the connection process within psycopg2 you might build the package with debug information and check if there are useful logging on connection time. On top of that you might want to enable the green mode and use the `wait_select()` function which might give you some extra logging. Instructions for both are in the documentation.

Another option is to try to connect using psycopg 3 and hack into its source to look for more logs. I can't remember if it still produces debug info on connection, which you might enable putting the `psycopg` logger to debug level and configuring a handler for it. For some time such code was there but it might have been yanked out because wasn't producing anything useful for debug in the use cases I have tested.

More likely, it's something that happens within the bowels of RDS and the client is not made aware of anything.
